transport/serialize/jsonserializer.go

```diff
@@ -46,6 +46,26 @@ func (s *JSONSerializer) Deserialize(data []byte) (wamp.Message, error) {
 	return listToMsg(wamp.MessageType(typ), v)
 }
 
+// SerializeDataItem encodes any object/structure into a json payload.
+func (s *JSONSerializer) SerializeDataItem(item interface{}) ([]byte, error) {
+	var b []byte
+	return b, codec.NewEncoderBytes(&b, jh).Encode(item)
```

Comment:
same

Reply:
Then I think I'll also change the `Serialize` method too, as it was a source for `SerializeDataItem`.
```go
func (s *JSONSerializer) Serialize(msg wamp.Message) ([]byte, error) {
	var b []byte
	return b, codec.NewEncoderBytes(&b, jh).Encode(msgToList(msg))
}
```

Reply:
Played a bit with this, https://go.dev/play/p/y-Qkwpj1FKl evaluates to 2, so it's right to left, but https://go.dev/ref/spec#Order_of_evaluation would suggest otherwise, unless I'm missing something.

Reply:
Found this https://github.com/golang/go/issues/23188 which is an interesting read. I would suggest to use this in the new code, which makes it more explicit:

```
func foo() ([]byte, error) {
	var b []byte
	err := b, codec.NewEncoderBytes(&b, jh).Encode(msgToList(msg))
	return b, err
}
```

and leave the old code as it is. FWIW you also have tests that demonstrate that this works, which should also catch any changes in case Go modifies the details of evaluation order.